avm/res/network/private-dns-zone/soa/main.bicep

```diff
@@ -0,0 +1,95 @@
+metadata name = 'Private DNS Zone SOA record'
+metadata description = 'This module deploys a Private DNS Zone SOA record.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent Private DNS zone. Required if the template is used in a standalone deployment.')
+param privateDnsZoneName string
+
+@description('Required. The name of the SOA record.')
+param name string
+
+@description('Optional. The metadata attached to the record set.')
+param metadata object?
+
+@description('Optional. A SOA record.')
+param soaRecord object = {}
```

Comment:
In other children you used nullable. Should we use it consistently?

Reply:
If that works that is. Sometimes it's an an array, sometimes an object

Reply:
can update was missed in my changes due to being an object not an array!